src/tests/QueryBuilderTests.cpp

```diff
@@ -418,6 +418,53 @@ TEST_CASE_METHOD(SqlTestFixture, "SqlQueryBuilder.WhereColumn", "[SqlQueryBuilde
                                   WHERE "left" = "right")"));
 }
 
+TEST_CASE_METHOD(SqlTestFixture,
+                 "Where: SqlQualifiedTableColumnName OP SqlQualifiedTableColumnName",
+                 "[SqlQueryBuilder]")
+{
+    checkSqlQueryBuilder(
+        [](SqlQueryBuilder& q) {
+            return q.FromTable("That")
+                .Select()
+                .Field("foo")
+                .Where(SqlQualifiedTableColumnName { .tableName = "That", .columnName = "left" },
+                       "=",
+                       SqlQualifiedTableColumnName { .tableName = "That", .columnName = "right" })
+                .All();
+        },
+        QueryExpectations::All(R"(SELECT "foo" FROM "That"
+                                  WHERE "That"."left" = "That"."right")"));
+}
+
+TEST_CASE_METHOD(SqlTestFixture,
+                 "Where: left IS NULL",
+                 "[SqlQueryBuilder]")
+{
+    checkSqlQueryBuilder(
+        [](SqlQueryBuilder& q) {
+            return q.FromTable("That")
+                .Select()
+                .Field("foo")
+                .Where("Left1", SqlNullValue)
+                .Where("Left2", std::nullopt)
+                .All();
+        },
+        QueryExpectations::All(R"(SELECT "foo" FROM "That"
+                                  WHERE "Left1" IS NULL AND "Left2" IS NULL)"));
+
+    checkSqlQueryBuilder(
+        [](SqlQueryBuilder& q) {
+            return q.FromTable("That")
+                .Select()
+                .Field("foo")
+                .Not().Where("Left1", SqlNullValue)
```

Comment:
I think this is not the most intuitive API, something like 
```suggestion
                .Where("Left1").Not(SqlNullValue)
```
is better, and can distinguish between complete/incomplete builder state

Reply:
I agree. There is also `WhereNot(value)` though. We could add a `WhereNotEquals(a, b)` express inequality.

`Where(a).Not(b)` I fear, might be hard to read too. We should focus on more expressive function member names instead. We already have `WhereNull` and `WhereNotNull` for example.

Reply:
so we can hide `Where(a).Not(b)` as implementation detail and provide `WhereNotEquals(a,b)` for the user, what do you think?

Reply:
@Yaraslaut I've been adding `WhereNotEqual(a, b)` and adapted the tests accordingly. There is no need to add `Where(a).Not(b)` as an implementation detail. (I also cannot think of a way to implement Where(a).Not(b) without breaking current Where(a) semantics).

P.s.: I was generally getting inspired on the query builder API by the Dapper project: https://github.com/DapperLib/Dapper :)